panel/src/components/Forms/Writer/Toolbar.vue

```diff
@@ -297,6 +304,25 @@ export default {
 				click: () => this.command(entry.command ?? type)
 			};
 		},
+		/**
+		 * Helper method to expand node types
+		 * @returns {Array}
+		 */
+		expandNodeTypes(nodes) {
+			const expanded = [];
+
+			nodes.forEach((node) => {
+				if (node === 'heading') {
+					this.headings.forEach((level) => {
+						expanded.push(`h${level}`);
+					});
+				} else {
+					expanded.push(node);
+				}
+			});
+
+			return expanded;
+		},
```

Comment:
`Todo:` I do not think we need to expand the headings here. The `Heading` node itself is registering multiple buttons depending on the heading level. Did this not work for you or why did you add this new helper method?

Reply:
It is set as `heading` index in the `nodes` array. But in the `available` array, it is passed as "h1-h6" indexes instead of `heading`. So I had to convert `heading` to `h*` indexdes.

Reply:
But something is off here, the headings node entries show up in different order before/after the PR. I am not sure I can fully follow yet why this change is needed.